### PR TITLE
Have DB creation script account for username

### PR DESCRIPTION
**Pull request by drewmarshburn**
This script previously made incorrect assumptions about the name of the database user and the name of the database. Providing connection strings with no db name initiates a connection using the username as the database name. Instead, the connection should always be made to the default `postgres` database.

### CODE REVIEW

File: deploy/charts/firefly/scripts/ff-db-migrations.sh [hunk 1]
```diff
@@ -3,29 +3,34 @@
...
+echo "Provided connection string: '${PSQL_URL}'"
+
 # Extract the database name from the end of the PSQL URL, and check it's there
-DB_NAME=`echo ${PSQL_URL} | sed 's/^.*\///'`
-COLONS=`echo -n $DB_NAME | sed 's/[^:]//g'`
+DB_PARAMS=`echo ${PSQL_URL} | sed 's!^.*/!!'`
+DB_NAME=`echo ${DB_PARAMS} | sed 's!?.*!!'`
 echo "Database name: '${DB_NAME}'"
+USER_NAME=`echo ${PSQL_URL} | sed 's!^.*//!!' | sed 's!:.*$!!'`
+echo "Username: '${USER_NAME}'"
```
Comment (hfuss) [inline deploy/charts/firefly/scripts/ff-db-migrations.sh:13]: Does this still work if the DB doesn't require auth? Know thats a bit unlikely / harder to test.

**Review by hfuss — APPROVED** [review]
lgtm otherwise! Thanks for finding this edge case and submitting a fix.

File: deploy/charts/firefly/scripts/ff-db-migrations.sh (hunk 1 above)
Comment (drewmarshburn) [reply deploy/charts/firefly/scripts/ff-db-migrations.sh:13]: Yeah that's a good point. Now, we need connection strings in at least this format:

"postgresql://[username]:[password]/[database]"

Looking at their doc (https://www.postgresql.org/docs/current/libpq-connect.html), supporting all formats of connection strings would be pretty difficult. Wondering what we need to support here? @peterbroadhurst and @hfuss  any insights from what you've previously deployed? It may also be easier to provide the PSQL connection info as separate chart values instead of winding up with all this `sed` `echo fun | sed 's/fun/suffering/'`.

**Comment by hfuss** [comment]
@drewmarshburn please rebase your PR off the latest upstream/main to get the changes from #359 and to retrigger the Go build that broke yesterday.

This should be good to merge after that.

File: deploy/charts/firefly/scripts/ff-db-migrations.sh (hunk 1 above)
Comment (hfuss) [reply deploy/charts/firefly/scripts/ff-db-migrations.sh:13]: I think assuming the PSQL string will have basic auth creds is fine for now. Most users are going to have a setup like that when using the chart.

**Review by nguyer — APPROVED** [review]
